### PR TITLE
Add `CoordsIter::exterior_coords_iter`

### PR DESCRIPTION
**Pull request by frewsxcv**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

This will get incorporated into https://github.com/georust/geo/pull/592

Fixes https://github.com/georust/geo/issues/542

~Based on and blocked by https://github.com/georust/geo/pull/593~

### CODE REVIEW

**Comment by frewsxcv** [comment]
Just merged in master, so this is ready for review

File: geo/src/algorithm/coords_iter.rs [hunk 1]
```diff
@@ -313,6 +411,31 @@ impl<'a, T: CoordinateType + 'a> CoordsIter<'a> for Geometry<T> {
             Geometry::Triangle(g) => g.coords_count(),
         }
     }
+
+    fn exterior_coords_iter(&'a self) -> Self::ExteriorIter {
+        match self {
```
Comment (michaelkirk) [inline geo/src/algorithm/coords_iter.rs:416]: Sometimes I wonder if we should adopt https://docs.rs/enum_dispatch/0.3.4/enum_dispatch/

File: geo/src/algorithm/coords_iter.rs [hunk 2]
```diff
@@ -329,14 +452,42 @@ pub struct MapCoordsIter<
...
+impl<'a, T: 'a + CoordinateType, Iter1: Iterator<Item = &'a Iter2>, Iter2: CoordsIter<'a>> Iterator
+    for MapCoordsIter<'a, T, Iter1, Iter2>
 {
     type Item = Iter2::Iter;
 
     fn next(&mut self) -> Option<Self::Item> {
         self.0.next().map(|g| g.coords_iter())
     }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
```
Comment (michaelkirk) [inline geo/src/algorithm/coords_iter.rs:464]: 👀 good catch

**Review by michaelkirk — APPROVED** [review]
nice!

bors r+

**Pull request merged**